ext/standard/file.c

```diff
@@ -566,6 +566,11 @@ PHP_FUNCTION(file_get_contents)
 		RETURN_FALSE;
 	}
 
+	/* disabling the read buffer allows doing the whole transfer
+	   in just one read() system call */
+	if (php_stream_is(stream, PHP_STREAM_IS_STDIO))
```

Comment:
@MaxKellermann I don't quite understand why this pull request got closed in it's entirety. I know there is an issue around commits 2+3, but commit 1 (file_get_contents without read buffer) is a clear, simple and non-breaking and uncontroversial improvement IMO. Maybe commit #e45f829 can be extracted into a new pull request to get merged?

Reply:
OK here you are https://github.com/php/php-src/pull/8547